### PR TITLE
Add monaco editor behind feature flag

### LINKED ISSUE #15635: [Monaco] - Create feature flagged DE with Monaco instead of code-mirror
- [ ] Use https://www.npmjs.com/package/@monaco-editor/react react wrapper. 

- [ ] test ability to do syntax highlighting

- [x] snippets

- [x] themes

- [ ] glyphs on margins

### PR DESCRIPTION
**Pull request by ebb-tide**
Closes #15635 

Kapture 2019-10-30 at 12 04 57

### CODE REVIEW

File: ui/src/shared/components/FluxMonacoEditor.tsx [hunk 1]
```diff
@@ -0,0 +1,18 @@
+// Libraries
+import React, {FC} from 'react'
+import Editor from '@monaco-editor/react'
+import 'src/external/monaco.flux'
```
Comment (drdelambre) [inline ui/src/shared/components/FluxMonacoEditor.tsx]: 🔥

File: ui/src/external/monaco.flux.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
+import initialize from 'src/external/monaco'
+
+initialize().then((monaco: any) => {
+  monaco.languages.register({id: 'Flux'})
+
+  monaco.languages.setMonarchTokensProvider('Flux', {
```
Comment (drdelambre) [inline ui/src/external/monaco.flux.ts]: is this what we connect to wasm later?

File: ui/src/external/monaco.flux.ts [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    },
+  })
+
+  monaco.editor.defineTheme('FluxTheme', {
+    base: 'vs',
+    inherit: false,
+    rules: [{token: 'keyword', foreground: 'ff00ff', fontStyle: 'bold'}],
+  })
+
+  monaco.languages.registerCompletionItemProvider('Flux', {
```
Comment (drdelambre) [inline ui/src/external/monaco.flux.ts]: 🔥🔥🔥🔥🔥🔥🔥🔥🔥🔥🔥🔥🔥🔥🔥🔥🔥🔥🔥🔥🔥🔥🔥🔥🔥🔥[ might wanna pull this out into it's own file too, but until then... ] 🔥🔥🔥🔥🔥🔥🔥🔥🔥🔥🔥🔥🔥🔥🔥🔥🔥🔥🔥🔥🔥🔥🔥🔥🔥🔥🔥🔥🔥🔥

File: ui/src/external/monaco.flux.ts (hunk 2 above)
Comment (ebb-tide) [reply ui/src/external/monaco.flux.ts]: not exactly. these are all for static syntax things. We pass props to Editor for things that we'll get from wasm

**Pull request closed without merging**